packages/flutter_tools/lib/src/build_system/targets/localizations.dart

```diff
@@ -0,0 +1,61 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter_tools/src/artifacts.dart';
+import 'package:flutter_tools/src/base/io.dart';
+import 'package:flutter_tools/src/base/logger.dart';
+import 'package:meta/meta.dart';
+import 'package:process/process.dart';
+
+import '../../base/file_system.dart';
+import '../build_system.dart';
+
+Future<void> generateLocalizations({
+  @required String flutterRoot,
+  @required FileSystem fileSystem,
+  @required ProcessManager processManager,
+  @required Artifacts artifacts,
+  @required Logger logger,
+}) async {
+  final String genL10nPath = fileSystem.path.join(
+    flutterRoot,
+    'dev',
+    'tools',
+    'localization',
+    'bin',
+    'gen_l10n.dart',
+  );
+  final ProcessResult result = await processManager.run(<String>[
+    artifacts.getArtifactPath(Artifact.engineDartBinary),
```

Comment:
Also: deferred localizations - are there any downsides to always enabling for web builds?

Reply:
Also, turning off deferred loading results in the messages for a locale to load synchronously (via a SynchronousFuture), which might affect how the app starts up

Reply:
As Shi Hao said, when I implemented it I thought it should be up for the developer to decide as its not an optimisation for small locale files, as it will require a second download for the locales on start up. 

If there are already too many options for developers, and you want to cut down on options, I do not think there is too much harm in always keeping it on for the web though.

Reply:
If we delegate to a config file its not as much of a problem for now. Longer term we need to solve the issue of generating code into the project source directory.